src/goto-instrument/code_contracts.cpp

```diff
@@ -242,6 +262,82 @@ void code_contractst::code_contracts(
       apply_contract(goto_function.body, it);
 }
 
+goto_programt code_contractst::make_ensures_assertion_for_call(
+  const code_function_callt &call)
+{
+  goto_programt goto_assertion;
+
+  // we don't handle function pointers
+  if(call.function().id() != ID_symbol)
+    return goto_assertion;
+
+  const irep_idt &function = to_symbol_expr(call.function()).get_identifier();
+  const symbolt &f_sym = ns.lookup(function);
+  const code_typet &type = to_code_type(f_sym.type);
+
+  exprt ensures = static_cast<const exprt &>(type.find(ID_C_spec_ensures));
+
+  // is there a contract?
+  if(ensures.is_nil())
+    return goto_assertion;
```

Comment:
This intends `return None`, but is vulnerable to accidentally returning a partial program if someone moves some prep work to the top of the function, e.g. allocating a temporary, or labelled SKIP or whatever. Suggest instead this function should return `optionalt<goto_programt>` so you can use `return {}` when you intend to report failure / no-action.